src/snapred/backend/recipe/FetchGroceriesRecipe.py

```diff
@@ -0,0 +1,152 @@
+from typing import Any, Dict, List
+
+from mantid.simpleapi import CloneWorkspace
+
+from snapred.backend.dao.ingredients.GroceryListItem import GroceryListItem
+from snapred.backend.dao.RunConfig import RunConfig
+from snapred.backend.log.logger import snapredLogger
+from snapred.backend.recipe.algorithm.FetchGroceriesAlgorithm import FetchGroceriesAlgorithm as FetchAlgo
+from snapred.meta.decorators.Singleton import Singleton
+
+logger = snapredLogger.getLogger(__name__)
+
+
+@Singleton
+class FetchGroceriesRecipe:
+    """
+    You need some data?  Yeah, I can get that for you.  Just send me a list.
+    """
+
+    def __init__(self):
+        self._loadedRuns: Dict[(str, str, bool), int] = {}
+
+    def _createFilenameFromRunConfig(self, runConfig: RunConfig) -> str:
+        if runConfig.isLite:
+            return f"{runConfig.IPTS}shared/lite/SNAP_{runConfig.runNumber}.lite.nxs.h5"
+        else:
+            return f"{runConfig.IPTS}nexus/SNAP_{runConfig.runNumber}.nxs.h5"
+
+    def _createNexusWorkspaceName(self, runConfig) -> str:
+        name: str = f"_TOF_RAW_{runConfig.runNumber}"
+        if runConfig.isLite:
+            name = name + "_lite"
+        return name
+
+    def fetchNexusData(self, runConfig: RunConfig, loader: str = "") -> Dict[str, Any]:
+        """
+        Fetch just a nexus data file
+        inputs:
+        - runConfig, a RunConfig object corresponf to the data desired
+        - loader, the loading algorithm to use, if you think you already know
+        outputsa dictionary with
+        - "result": true if everything ran correctly
+        - "loader": the loader that was used by the algorithm, use it next time
+        - "workspace": the name of the workspace created in the ADS
+        """
+
+        workspaceName = self._createNexusWorkspaceName(runConfig)
+        fileName = self._createFilenameFromRunConfig(runConfig)
+        data: Dict[str, Any] = {
+            "result": False,
+            "loader": loader,
+            "workspace": workspaceName,
+        }
+        thisKey = (runConfig.runNumber, runConfig.IPTS, runConfig.isLite)
+        if self._loadedRuns.get(thisKey, 0) > 0:
```

Comment:
what if its not already loaded but I would like to cache it, like in the case of DiffractionCalibration where I run it for the first time?

Reply:
The first time you call this, it will load the entire file.  The second time you call this, it will just clone the previous file.  What do you want the cache feature to do?  Keep a record that someone wants this object and hang onto it?

Reply:
It should do a load and clone.  If we used this as is for the DiffractionCalibration flow it would dirty the input workspace, which is unhelpful when we want to adjust parameters and try again.  It is quicker to clone the raw input and run it again than load the raw input again from disk.

Reply:
if there was a flag to load + clone that would suffice.

Reply:
ah, that makes sense

Reply:
I think that's what you were looking for